pysph/examples/rb_rotation_matrices/stack_of_cylinders.py

```diff
@@ -0,0 +1,239 @@
+        xc6, yc6 = (xc5 + self.cylinder_diameter + self.cylinder_spacing, yc5)
+
+        x_six_1, y_six_1 = np.concatenate(
+            (xc1, xc2, xc3, xc4, xc5, xc6)), np.concatenate((yc1, yc2, yc3,
+                                                             yc4, yc5, yc6))
+        x_six_2, y_six_2 = x_six_1, y_six_1 + 2. * self.cylinder_diameter
+        x_six_3, y_six_3 = x_six_2, y_six_2 + 2. * self.cylinder_diameter
+
+        x_six, y_six = np.concatenate(
+            (x_six_1, x_six_2, x_six_3)), np.concatenate((y_six_1, y_six_2,
+                                                          y_six_3))
+
+        # now create five cylinder rows
+        xc1, yc1 = get_2d_circle(self.cylinder_spacing, self.cylinder_radius, [
+            self.cylinder_diameter + self.cylinder_spacing,
+            3. * self.cylinder_radius + self.cylinder_spacing
+        ])
+        xc2, yc2 = (xc1 + self.cylinder_diameter + self.cylinder_spacing, yc1)
+        xc3, yc3 = (xc2 + self.cylinder_diameter + self.cylinder_spacing, yc2)
+        xc4, yc4 = (xc3 + self.cylinder_diameter + self.cylinder_spacing, yc3)
+        xc5, yc5 = (xc4 + self.cylinder_diameter + self.cylinder_spacing, yc4)
+
+        x_five_1, y_five_1 = np.concatenate(
+            (xc1, xc2, xc3, xc4, xc5)), np.concatenate((yc1, yc2, yc3, yc4,
+                                                        yc5))
+        x_five_2, y_five_2 = x_five_1, y_five_1 + 2. * self.cylinder_diameter
+        x_five_3, y_five_3 = x_five_2, y_five_2 + 2. * self.cylinder_diameter
+        x_five, y_five = np.concatenate((x_five_1, x_five_2,
+                                         x_five_3)), np.concatenate(
+                                             (y_five_1, y_five_2, y_five_3))
+
+        x, y = np.concatenate((x_six, x_five)), np.concatenate((y_six, y_five))
+
+        # create body_id
+        no_particles_one_cylinder = len(xc1)
+        total_bodies = 3 * 5 + 3 * 6
+
+        body_id = np.array([], dtype=int)
+        for i in range(total_bodies):
+            b_id = np.ones(no_particles_one_cylinder, dtype=int) * i
+            body_id = np.concatenate((body_id, b_id))
+
+        return x, y, body_id
+
+    def geometry(self):
+        import matplotlib.pyplot as plt
+        # please run this function to know how
+        # geometry looks like
+        xc, yc, body_id = self.create_cylinders_stack()
+        xt, yt = self.create_dam()
+        xw, yw = self.create_wall()
+
+        plt.scatter(xc, yc)
+        plt.scatter(xt, yt)
+        plt.scatter(xw, yw)
+        plt.axes().set_aspect('equal', 'datalim')
+        print("done")
+        plt.show()
+
+    def post_step(self, solver):
+        t = solver.t
+        dt = solver.dt
+        T = self.wall_time
+        if (T - dt / 2) < t < (T + dt / 2):
+            for pa in self.particles:
+                if pa.name == 'wall':
+                    pa.y += 14 * 1e-2
```

Comment:
Why this is done? Can you please add a reference to this too.

Reply:
Perhaps a comment here to explain the intent may be good?